include/iceflow/IceFlowPubSubBase.hpp

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2021 The IceFlow Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ndn-svs/security-options.hpp"
+#include "ndn-svs/svspubsub.hpp"
+#include "ndn-svs/svsync.hpp"
+#include "ringbuffer.hpp"
+
+#include "logger.hpp"
+#include <iostream>
+#include <thread>
+namespace iceflow {
+
+class IceFlowPubSub {
+
+public:
+  IceFlowPubSub(const std::string &syncPrefix, const std::string &topic,
+                const std::vector<int> &nTopic)
+      : mTopic(topic), topicPartition(nTopic.size()) {
+    // Use HMAC signing for Sync Interests
+    // Note: this is not generally recommended, but is used here for simplicity
+    ndn::svs::SecurityOptions secOpts(m_keyChain);
+    secOpts.interestSigner->signingInfo.setSigningHmacKey(
+        "dGhpcyBpcyBhIHNlY3JldCBtZXNzYWdl");
+
+    // Sign data packets using SHA256 (for simplicity)
+    secOpts.dataSigner->signingInfo.setSha256Signing();
+
+    // Do not fetch publications older than 10 seconds
+    ndn::svs::SVSPubSubOptions opts;
+    opts.UseTimestamp = true;
+    opts.MaxPubAge = ndn::time::milliseconds(10000);
```

Comment:
The CI is complaining about these:

```suggestion
    opts.useTimestamp = true;
    opts.maxPubAge = ndn::time::milliseconds(10000);
```